Overview:

Context:

There is no standardized method for assembling a genome, and sequencing depth can impact the 
accuracy and reliability of genomic analysis. Average nucleotide identity (ANI) and Mash scores 
are commonly used to assess genome similarity, yet the impact of sequencing depth on this metrics 
remains unclear. This project aims to evaluate how varying sequencing depths influence ANI and 
Mash scores, and give more clarity on their reliability under these different conditions. 
Understanding this is important, as it will provide researchers with better insight on the 
accuracy of their data and genomic comparisons.Create a reference genome assembly and generate 
multiple read sets at varying sequencing depths, with multiple random subsamples for each depth.



Goals: 

Create a reference genome assembly and various read sets at different depth coverages with 
multiple random samples generated of each 

Users will be able to assess ANI and mesh scores of read sets at different depth coverages in 
comparison with a reference genome assembly. 

success will be measured based on compared ANI and mash scores between different read sets of
depth coverage. Higher ANI percentage and a lower mash score indicate a higher assemby
similarity

Non-goals:

This project will not focus on optimizing genome assembly parameters past standard settings.

This project will not compare different genome assemblers. 

We will not be comparing the ANI and Mash scores to each other. 

Sequencing depth will not be assessed for evenness throughout the entire genome assembly.



Milestones:

